src/aggregation/bucket/filter.rs

```diff
@@ -0,0 +1,380 @@
+use std::fmt::Debug;
+
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+
+use crate::aggregation::agg_req_with_accessor::AggregationsWithAccessor;
+use crate::aggregation::intermediate_agg_result::{
+    IntermediateAggregationResult, IntermediateAggregationResults, IntermediateBucketResult,
+};
+use crate::aggregation::segment_agg_result::{
+    build_segment_agg_collector_with_reader, CollectorClone, SegmentAggregationCollector,
+};
+use crate::query::{Query, QueryParser, Weight};
+use crate::schema::Schema;
+use crate::tokenizer::TokenizerManager;
+use crate::{DocId, SegmentReader, TantivyError};
+
+/// Filter aggregation creates a single bucket containing documents that match a query.
+///
+/// # Usage
+/// ```rust
+/// use tantivy::aggregation::bucket::filter::FilterAggregation;
+/// use tantivy::query::TermQuery;
+///
+/// // Query strings are parsed using Tantivy's standard QueryParser
+/// let filter_agg = FilterAggregation::new("category:electronics AND price:[100 TO 500]".to_string());
+///
+/// // Direct Query objects can be used for custom query types
+/// let term_query = TermQuery::new(
+///     tantivy::Term::from_field_text(tantivy::schema::Field::from_field_id(0), "electronics"),
+///     tantivy::schema::IndexRecordOption::Basic
+/// );
+/// let filter_agg = FilterAggregation::new_with_query(Box::new(term_query));
+/// ```
+///
+/// # Result
+/// The filter aggregation returns a single bucket with:
+/// - `doc_count`: Number of documents matching the filter
+/// - Sub-aggregation results computed on the filtered document set
+#[derive(Debug, Clone)]
+pub struct FilterAggregation {
+    /// The query for filtering - can be either a query string or a direct Query object
+    query: FilterQuery,
+}
+
+/// Represents different ways to specify a filter query
+#[derive(Debug)]
+pub enum FilterQuery {
+    /// Query string that will be parsed using Tantivy's standard parsing facilities
+    /// Accepts query strings that can be parsed by QueryParser::parse_query()
+    QueryString(String),
+
+    /// Direct Query object for extension query types
+    /// This bypasses JSON parsing entirely for maximum performance
+    Direct(Box<dyn Query>),
```

Comment:
```suggestion
    CustomQuery(Box<dyn Query>),
```

For me both variants are direct, but this one allows to pass in custom queries

Reply:
That's a better name. Changed it to `CustomQuery`